src/rules/pyupgrade/rules/import_replacements_six.rs

```diff
@@ -0,0 +1,211 @@
+use std::collections::HashMap;
+
+use once_cell::sync::Lazy;
+use ruff_macros::derive_message_formats;
+use rustpython_ast::{AliasData, Located, Stmt};
+
+use crate::ast::types::Range;
+use crate::checkers::ast::Checker;
+use crate::define_violation;
+use crate::fix::Fix;
+use crate::registry::{Diagnostic, Rule};
+use crate::rules::pyupgrade::helpers::{get_fromimport_str, ImportFormatting};
+use crate::source_code::Locator;
+use crate::violation::AlwaysAutofixableViolation;
+
+define_violation!(
+    pub struct ImportReplacementsSix;
+);
+impl AlwaysAutofixableViolation for ImportReplacementsSix {
```

Comment:
Why is the `six` stuff a separate rule? Is that following a `pyupgrade` convention?

This is maybe controversial, but my instinct is to skip all the explicit Python 2-to-Python 3 conversion stuff (like `six` conversions). We have some of that already, but it feels really one-time, whereas much of the other `pyupgrade` stuff is useful for new Python 3 code that's being written on an ongoing basis (e.g., the `typing` import replacements, or the type annotation rewrites, or the string format rewrites). Those kinds of things come up as you write new Python 3 code. But you'd never really expect to accidentally import `six` in new Python 3 code.

Does that make sense? In other words: I'm proposing that we omit the rules from `pyupgrade` that you'd expect to be "one-time changes".

Reply:
Although now that I'm looking at the code, again wondering what the difference is between the responsibilities of this rule and `import_replacements.rs` :)

Reply:
I actually agree with this a lot. It does not seem wise to me to maintain code, when the user could go directly to pyupgrade once, and then use us from then on. Would you like me to remove all of `import_replacements_six.rs`?

Reply:
I split them up because they function slightly differently, and I thought it would be easier to split the code.

Reply:
Does this file _also_ do some `six` import replacement though? E.g., this file has `SIX_MOVES_TO_COLLECTIONS`. (Just trying to understand before I chime in further.)

Reply:
Yes, I just handled the two groups a little differently, since pyupgrade did. I can also remove all references to six in `import_replacements.rs`.

Reply:
Ok, sounds good. Let's do it, I think. We can also remove `src/rules/pyupgrade/rules/remove_six_compat.rs`, but lets do that separately, since we need to deprecate the actual check.

Reply:
(Do you want to make that change? I can also do it, I'm in this file now. Just LMK what you prefer.)

Reply:
I can get it done!